samples/base/Logger.cpp

```diff
@@ -88,14 +96,37 @@ namespace darwin {
             _name = name;
         }
 
+        bool Logger::setFilePath(std::string const& filepath) {
+            bool ret = true;
+
+            std::ofstream testfile = std::ofstream(filepath, std::ios::out | std::ios::app);
+
+            if (!testfile.is_open()) {
+                std::clog << "Can't open log file " << filepath << std::endl;
+                ret = false;
+            } else {
+                _filepath = filepath;
+                _file.swap(testfile);
+            }
+
+            if (testfile.is_open())
+                testfile.close();
+
+            return ret;
+        }
+
         void Logger::RotateLogs() {
-            if (access(DARWIN_LOG_FILE, F_OK) != 0) {
+            if (access(_filepath.c_str(), F_OK) != 0) {
                 _fileMutex.lock();
```

Comment:
best practice to use RAII with `std::lock_guard lock(_fileMutex)` and let it be destroyed by the end of the scope

Reply:
good call (bis)

Reply:
change made, I leave you to comment/resolve the conversation